src/coreclr/src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1306,18 +1702,35 @@ internal static bool IsAttributeDefined(RuntimeModule decoratedModule, int decor
                     }
                     catch (Exception e)
                     {
-                        throw new CustomAttributeFormatException(
-                            SR.Format(isProperty ? SR.RFLCT_InvalidPropFail : SR.RFLCT_InvalidFieldFail, name), e);
+                        ex = e;
+                    }
+
+                    if (ex is not null)
+                    {
+                        RethrowException(name, isProperty, ex);
                     }
                 }
 
                 if (blobStart != blobEnd)
                 {
-                    throw new CustomAttributeFormatException();
+                    ThrowIncorrectData();
                 }
 
                 attributes.Add(attribute);
             }
+
+            [DoesNotReturn]
+            [StackTraceHidden]
+            static void ThrowInvalidPropertyException(string name) => throw new CustomAttributeFormatException(SR.Format(SR.RFLCT_InvalidPropFail, name));
```

Comment:
Are these throw helpers really improving anything? The method that this is called from is pretty big.

Reply:
Was to attempt to reduce loop size; but no it doesn't, the Jit generates the same code (just a jmp), other than also adding a huge chunk to the end of the method, but as you point out the method is already large, so likely doesn't change much and that should probably be handled by something in the Jit like "partially jitted methods" https://github.com/dotnet/runtime/pull/34522 or "hot/cold splitting" https://github.com/dotnet/runtime/issues/10412 instead?

Will revert these.